src/main/java/net/snowflake/ingest/connection/SecurityManager.java

```diff
@@ -176,4 +198,82 @@ String getToken()
     return token.get();
   }
 
+  /**
+   * Hashes input bytes using SHA-256.
+   *
+   * @param input  input bytes
+   * @param offset offset into the output buffer to begin storing the digest
+   * @param len    number of bytes within buf allotted for the digest
+   * @return SHA-256 hash
+   */
+  public static byte[] sha256Hash(byte[] input, int offset, int len)
+  {
+    try
+    {
+      MessageDigest md = MessageDigest.getInstance("SHA-256");
+      md.update(input, offset, len);
+      return md.digest();
+    }
+    catch (NoSuchAlgorithmException e)
+    {
+      throw new RuntimeException(e);
+    }
+  }
+  /**
+   * Hashes input bytes using SHA-256.
+   *
+   * @param input input bytes
+   * @return SHA-256 hash
+   */
+  public static byte[] sha256Hash(byte[] input)
+  {
+    return sha256Hash(input, 0, input.length);
+  }
+
+  /**
+   * Hashes input bytes using SHA-256 and converts hash into a string using
+   * Base64 encoding.
+   *
+   * @param input input bytes
+   * @return Base64-encoded SHA-256 hash
+   */
+  public static String sha256HashBase64(byte[] input)
```

Comment:
Method sha256HashBase64/sha256Hash is better moved to a separate utility class (maybe under util package namespace), like `Cryptor` so that it can be reused by other classes in the future.